Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -225,7 +225,7 @@ internal class CustomerIOImplementation: CustomerIOInstance {
         // we can reference the token and register it to a new profile.
         globalDataStore.pushDeviceToken = deviceToken
 
-        guard let identifier = profileStore.identifier else {
+        guard let identifier = profileStore.identifier, !identifier.isEmpty else {
             logger.info("no profile identified, so not registering device token to a profile")
             return
         }
```

Comment:
```suggestion
        guard let identifier = profileStore.identifier else {
            logger.info("no profile identified, so not registering device token to a profile")
            return
        }
        guard !identifier.isEmpty else {
            logger.error("profiles cannot be created without an identifier. Empty string was provided as the identifier, therefore, request to identify profile in the SDK is ignored. Try identifying a profile again with a non-empty identifier value.")
            return
        }
```

The message logged to the customer when they provide an empty value would be, "no profile identified, so not registering device token to a profile" which is not accurate to them. The customer did identify a profile, but the identifier is an invalid value. 

Therefore, I suggest we create a 2nd guard statement with a more helpful error message logged. 

I also used `logger.error` in hopes the customer notices the log message because it's an error log level.